core/src/main/java/io/grpc/MethodDescriptor.java

```diff
@@ -215,6 +216,24 @@ public InputStream streamResponse(RespT response) {
   }
 
   /**
+   * Returns whether this method is idempotent.
+   */
+  public boolean idempotent() {
+    return idempotent;
+  }
+
+  /**
+   * Set idempotency on this method.
+   *
+   * @param idempotent the idempotency of this method.
+   * @return this MethodDescriptor.
+   */
+  public MethodDescriptor idempotent(boolean idempotent) {
```

Comment:
Oh, you modify the `MethodDescriptor`. How about we make a copy with the modification in place.